agot-bg-game-server/src/client/MapComponent.tsx

```diff
@@ -131,19 +132,24 @@ export default class MapComponent extends Component<MapComponentProps> {
 
         const order = actionGameState.ordersOnBoard.get(r);
 
-        return this.renderOrder(r, order, orderImages.get(order.type.id));
+        return this.renderOrder(r, order, orderImages.get(order.type.id), true);
     }
 
-    renderOrder(region: Region, order: Order| null, backgroundUrl: string): ReactNode {
+    renderOrder(region: Region, order: Order| null, backgroundUrl: string, isActionGameState: boolean): ReactNode {
         return (
-            <div className={classNames("order-container", "hover-weak-outline", {"medium-outline hover-strong-outline clickable": order ? this.shouldHighlightOrder(region, order) : false})}
-                 style={{left: region.orderSlot.x, top: region.orderSlot.y}}
-                 onClick={() => order ? this.onOrderClick(region, order) : undefined}
-                 key={"region-" + region.id}
+            <div className={classNames("order-container", "hover-weak-outline", { "medium-outline hover-strong-outline clickable": order ? this.shouldHighlightOrder(region, order) : false })}
+                style={{ left: region.orderSlot.x, top: region.orderSlot.y }}
+                onClick={() => order ? this.onOrderClick(region, order) : undefined}
+                key={"region-" + region.id}
             >
-                <div className="order-icon" style={{
-                    backgroundImage: `url(${backgroundUrl})`
-                }}/>
+                {isActionGameState && <OverlayTrigger overlay={
+                    <Tooltip id={"order-owner"}>
+                        {this.getController(region)}
+                    </Tooltip>
+                } delay={{ show: 750, hide: 250 }}>
+                    <div className="order-icon" style={{ backgroundImage: `url(${backgroundUrl})` }} />
+                </OverlayTrigger>}
+                {!isActionGameState && <div className="order-icon" style={{ backgroundImage: `url(${backgroundUrl})` }} />}
```

Comment:
You can use `ConditionalWrap` to conditionnaly wrap a component by an other. This will remove the repeated `<div className="order-icon" ...>`.

Reply:
Thanks for the hint. New commit is coming soon.